controllers/binding/snapshotenvironmentbinding_controller.go

```diff
@@ -89,10 +90,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}
 	logger = logger.WithApp(*application)
 
-	snapshot, err := r.getSnapshotFromSnapshotEnvironmentBinding(ctx, snapshotEnvironmentBinding)
+	var snapshot *applicationapiv1alpha1.Snapshot
+	err = retry.OnError(retry.DefaultRetry, func(_ error) bool { return true }, func() error {
+		snapshot, err = r.getSnapshotFromSnapshotEnvironmentBinding(ctx, snapshotEnvironmentBinding)
+		return err
+	})
 	if err != nil {
-		logger.Error(err, "Failed to get Snapshot from the SnapshotEnvironmentBinding")
-		return ctrl.Result{}, err
+		return logger.HandleLoaderError(err, "Snapshot", "SnapshotEnvironmentBinding")
```

Comment:
Logger shouldn't have responsibility to decide code flow (yes there is exception in go, for `panic` and `fatal` but that ends everything)

https://github.com/redhat-appstudio/integration-service/pull/329